### ISSUE TITLE
 CBA_fnc_players might still be useful

### ISSUE THREAD

**Issue by Cuel** [CBATeam/CBA_A3#353]
```
allPlayers - The command also returns all connected headless clients.
_justPlayers = allPlayers - entities "HeadlessClient_F"
``` 

This is a discussion if getting back all headless clients is an expected output when using this command. 

I'd imagine that in most scenarios one would **not** want to run some code on HC's. 
My suggestion is to add back CBA_fnc_players and filter out headless clients before returning the array. Perhaps also check if the virtual entity spectators are included here.

If one wants the HC's, he/she can run the command
`entities "HeadlessClient_F"`

There's also another comment that I find weird:
```
NOTE: In player hosted game, the complete array of allPlayers may get delayed at the start
Use BIS_fnc_listPlayers if you need it earlier
```

**Comment by jonpas** [comment]
Or a param to return HCs or not.

**Comment by commy2** [comment]
Does `BIS_fnc_listPlayers` report headless clients?

**Comment by SzwedzikPL** [comment]
> Does BIS_fnc_listPlayers report headless clients?

`(allUnits + allDead) select {isPlayer _x}`

**Comment by commy2** [comment]
That doesn't answer my question. Does `isPlayer` report `false` for headless clients? That is not documented on the wiki.

**Comment by Cuel** [comment]
afaik `isPlayer` is true for HC

**Comment by SzwedzikPL** [comment]
Yes, i just checked. `isPlayer` is true for headless.

**Comment by commy2** [comment]
So what we need is:
`CBA_fnc_players`:
(allUnits + allDead) select {isPlayer _x && {_x isKindOf "HeadlessClient_F"}}

**Comment by Cuel** [comment]
`allPlayers select {!(_x isKindOf "HeadlessClient_F")}}` should be enough?

**Comment by commy2** [comment]
possibly. I'm confused about that note about local hosted MP

**Comment by Cuel** [comment]
I agree. No idea what it means. Does it delay the script or can it report an empty array at start?

**Comment by commy2** [comment]
Empty array most likely.

**Comment by Cuel** [comment]
Closed with #371
